sphinx/source/recovery.rst

```diff
@@ -1,105 +1,122 @@
+    --rpc-address=node2_rpc_ip:node2_rpc_port --ca=/path/to/new/network/certificate
+    vote --accept --proposal-id=proposal_id
 
-Once a quorum of members (defined by the constitution rules but typically, a majority of members) have agreed to recover the network, the network secrets are unsealed and the recovery of the private entries of the ledger is automatically started.
+Once a quorum of members (defined by the :ref:`governance` rules but typically, a majority of members) have agreed to recover the network, the network secrets are unsealed and the recovery of the private entries of the ledger is automatically started.
 
-.. note:: While the primary and all active backups are recovering the private ledger, no new transaction can be executed by the network.
+.. note:: While all nodes are recovering the private ledger, no new transaction can be executed by the network.
 
 .. mermaid::
 
     sequenceDiagram
         participant Members
-        participant Primary
-        participant Backup
+        participant Users
+        participant Node 2
+        participant Node 3
 
-        Members->>+Primary: Propose recovery + sealed network secrets
+        Members->>+Node 2: Propose recovery + sealed network secrets
+        Node 2-->>Members: Proposal ID
         loop Wait until quorum
-            Members->>+Primary: Vote(s)
+            Members->>+Node 2: Vote(s) for Proposal ID
         end
+        Note over Node 2: Proposal completes successfully
 
-        Primary->>+Primary: Initiate end of recovery protocol
+        Note over Node 2: Reading Private Ledger...
+        Note over Node 3: Reading Private Ledger...
 
-        Primary->>+Primary: Recover Private Ledger
-        Backup->>+Backup: Recover Private Ledger
+        Note over Node 2: Part of Network
+        Note over Node 3: Part of Network
 
-        Note over Primary: Part of Private Network
-        Note over Backup: Part of Private Network
+        loop Business transactions
+            Users->>+Node 2: JSON-RPC Request
+            Node 2-->>Users: JSON-RPC Response
+            Users->>+Node 3: JSON-RPC Request
+            Node 3-->>Users: JSON-RPC Response
+        end
 
 Once the recovery of the private ledger on all the nodes that have joined the new network is complete, the ledger is fully recovered and users are able to continue issuing business transactions.
 
 .. warning:: After recovery, the identity of the network has changed. The new network certificate ``networkcert.pem`` returned in :ref:`Phase 1: Crash-fault tolerant public network` needs to be distributed to all existing and new users.
 
 .. rubric:: Footnotes
 
+.. [#crash] When using Raft as consensus algorithm, CCF tolerates up to `N/2 - 1` crashed nodes (where `N` is the number of nodes constituing the network) before having to perform catastrophic recovery. For example, in a 5-node network, no more than 2 nodes are allowed to fail.
```

Comment:
constitu**t**ing


We generally talk about `f` failures rather than `N` nodes, though I agree that N is more intuitive for the layman. Can we append a sentence here along the lines of "This can also be phrased as Raft requiring at least `2f + 1` nodes in a network to tolerate up to `f` crashed nodes. If the network should survive 3 nodes failing, then there must be at least 7 nodes in the network"?

Reply:
I tried to mimic what the Raft paper says: "_They are fully functional (available) as long as any
majority of the servers are operational and can communicate with each other and with clients. Thus, a typical cluster of five servers can tolerate the failure of any two servers. Servers are assumed to fail by stopping; they may later recover from state on stable storage and rejoin the cluster._"

I think I prefer talking about `N` rather than `f` and as you said, I think it makes more sense from a user point of view.